### PR TITLE
[RShaman] Shadowlands stuff

### PR DESCRIPTION
**Pull request by niseko**
Most of this PR is cleanup & event listeners. Only things that really got touched are the EarthShield file and the tidal wave tracking move out of the ability tracker into the tidalwave file, with some better logic. Also a few js -> ts again.

### CODE REVIEW

File: src/common/SPELLS/talents/shaman.js [hunk 1]
```diff
@@ -56,6 +54,7 @@ export default {
   ASCENDANCE_TALENT_ELEMENTAL: { id: 114050, name: 'Ascendance', icon: 'spell_fire_elementaldevastation' },
 
   // Restoration
+  // TODO Surge of Earth
```
Comment (abelito75) [inline src/common/SPELLS/talents/shaman.js:57]: Might want to just add it but i won't stop the PR bc of it

File: src/parser/core/CASTS_THAT_ARENT_CASTS.js [hunk 2]
```diff
@@ -43,4 +44,5 @@ export default [
   SPELLS.WILD_MARK.id, //The debuff applied from enemies inside Wild Spirits
   //endregion
   //endregion
+  324748, //Celestial Guidance enchant proc
```
Comment (abelito75) [inline src/parser/core/CASTS_THAT_ARENT_CASTS.js]: Should add this to enchant.js

File: src/parser/shaman/restoration/modules/features/MasteryEffectiveness.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ class MasteryEffectiveness extends Analyzer {
   masteryHealEvents = [];
 
   on_byPlayer_heal(event) {
```
Comment (abelito75) [inline src/parser/shaman/restoration/modules/features/MasteryEffectiveness.js:30]: Convert to action listener... we hate magic functions

File: src/parser/shaman/restoration/modules/spells/EarthShield.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+class EarthShield extends EarthShieldCore {
+  static dependencies = {
+    ...EarthShieldCore.dependencies,
+    cooldownThroughputTracker: CooldownThroughputTracker,
+  };
+
+  protected cooldownThroughputTracker!: CooldownThroughputTracker;
+
+  getFeeding() {
+    return this.cooldownThroughputTracker.getIndirectHealing(SPELLS.EARTH_SHIELD_HEAL.id);;
```
Comment (abelito75) [inline src/parser/shaman/restoration/modules/spells/EarthShield.ts]: has 2 semicolons for no reason

File: src/parser/shaman/shared/talents/EarthShield.tsx [hunk 5]
```diff
@@ -57,33 +65,38 @@ class EarthShield extends Analyzer {
...
   }
 
+  // gets overriden by the rshaman module
+  getFeeding() {
+    return 0;
+  }
+
   statistic() {
     return (
       <StatisticBox
```
Comment (abelito75) [inline src/parser/shaman/shared/talents/EarthShield.tsx:90]: Convert to Statistic

File: src/parser/shaman/shared/talents/SpiritWolf.tsx [hunk 6]
```diff
@@ -33,7 +39,6 @@ class SpiritWolf extends Analyzer {
   }
 
   statistic() {
-
     return (
       <StatisticBox
```
Comment (abelito75) [inline src/parser/shaman/shared/talents/SpiritWolf.tsx:43]: Convert to Statistic

**Review by abelito75 — CHANGES_REQUESTED** [review]
Looks almost done! Great work!

File: src/parser/shaman/restoration/modules/features/MasteryEffectiveness.js (hunk 3 above)
Comment (niseko) [reply src/parser/shaman/restoration/modules/features/MasteryEffectiveness.js:30]: doing it whenever i'm converting this to .ts

Comment (abelito75) [reply src/parser/shaman/restoration/modules/features/MasteryEffectiveness.js:30]: Okay

**Pull request merged**